### PR TITLE
Clarify EventRef's data parameter is triggerEventRef's payload

### PR DESCRIPTION
**Pull request by LCaparelli**
Signed-off-by: Lucas Caparelli <lucas.caparelli112@gmail.com>

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [X] Specification
- [X] Schema
- [ ] Examples
- [ ] Usecases
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**: Clarify that [EventRef](https://github.com/serverlessworkflow/specification/blob/master/specification.md#eventref-definition)'s `triggerEventRef` receives the EventRef's `data` parameter as its payload.

**Special notes for reviewers**:

**Additional information (if needed):** Suggestion of change from #161 discussion

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

File: specification.md [hunk 1]
```diff
@@ -1225,7 +1225,7 @@ function. They can include either static values or reference the states data inp
 | --- | --- | --- | --- |
 | [triggerEventRef](#Event-Definition) | Reference to the unique name of a `produced` event definition | string | yes |
 | [resultEventRef](#Event-Definitions) | Reference to the unique name of a `consumed` event definition | string | yes |
-| data | If string type, an expression which selects parts of the states data output to become the data (payload) of the produced event. If object type, a custom object to become the data (payload) of trigger/produced event. | string or object | no |
+| data | If string type, an expression which selects parts of the states data output to become the data (payload) of the event referenced by `triggerEventRef`. If object type, a custom object to become the data (payload) of trigger/produced event. | string or object | no |
```
Comment (tsurdilo) [inline specification.md:1228]: The second sentence ("If object type, a custom object to become the data (payload) of trigger/produced event.") could also be updated to have your change - "event referenced by triggerEventRef".

Comment (tsurdilo) [inline specification.md:1228]: Please also update the "description" param  in the json schemas for this parameter to match the one in the specification document.

Comment (tsurdilo) [inline specification.md:1228]: Another thing, look at the documentation below this description in the spec section to make sure that it fits this updated description as well. thanks.

**Review by tsurdilo — CHANGES_REQUESTED** [review]

File: specification.md (hunk 1 above)
Comment (LCaparelli) [reply specification.md:1228]: I tried grepping the schema directory for the old description, but it returned no results:

```
╰─ grep 'If string type, an expression which selects parts of the states data output to become the data (payload) of the produced event' -ir schema/* -c
schema/common.json:0
schema/events.json:0
schema/extensions/kpi.json:0
schema/functions.json:0
schema/workflow.json:0
```
Turns out the description was not the same on the schema, which is why I missed it at first:

```
 379           "description": "If String, expression which selects parts of the states data output to become the data of the produced event. If object a custom object to become the data of produced event."
```

Will update this in a bit, thanks! :-)

File: specification.md [hunk 2]
```diff
@@ -3915,4 +3915,4 @@ You can find a list of other languages, technologies and specifications related
 ## License
 
 Serverless Workflow specification operates under the
-[Apache License version 2.0](LICENSE).
\ No newline at end of file
+[Apache License version 2.0](LICENSE).
```
Comment (tsurdilo) [inline specification.md:3918]: Wonder why both of your PRs include this change? Maybe some newline char issue?

**Review by tsurdilo — APPROVED** [review]

File: specification.md (hunk 2 above)
Comment (LCaparelli) [reply specification.md:3918]: Yeah, it's just an additional line break vim added automatically to the end of the file.

Using `od` we can see the master branch has no line break:

```
╰─ od -c specification.md
# output omitted
0373300       W   o   r   k   f   l   o   w       s   p   e   c   i   f
0373320   i   c   a   t   i   o   n       o   p   e   r   a   t   e   s
0373340       u   n   d   e   r       t   h   e  \n   [   A   p   a   c
0373360   h   e       L   i   c   e   n   s   e       v   e   r   s   i
0373400   o   n       2   .   0   ]   (   L   I   C   E   N   S   E   )
0373420   .
0373421
```

From this branch:

```
╰─ od -c specification.md
# output omitted
0373340   m   d   )   .  \n  \n   #   #       L   i   c   e   n   s   e
0373360  \n  \n   S   e   r   v   e   r   l   e   s   s       W   o   r
0373400   k   f   l   o   w       s   p   e   c   i   f   i   c   a   t
0373420   i   o   n       o   p   e   r   a   t   e   s       u   n   d
0373440   e   r       t   h   e  \n   [   A   p   a   c   h   e       L
0373460   i   c   e   n   s   e       v   e   r   s   i   o   n       2
0373500   .   0   ]   (   L   I   C   E   N   S   E   )   .  \n
0373516
```

**Review by manuelstein — APPROVED** [review]
LGTM

**Pull request merged**